docs/constraint_creation.md

```diff
@@ -0,0 +1,44 @@
+# How Constraints are Created
```

Comment:
Thanks for splitting this up, I think it makes sense to start breaking apart the `README` at this point 👍